### DISCUSSION TITLE
Error install 1.21.0 on macOS

### DISCUSSION THREAD

**Discussion by Doc-Acula** [Acly/krita-ai-diffusion#996] [General]
Hi there,
i just updated to 1.21.0 on macOS, but I get the error message in Krita: Server running - Connection error: Connectio established, but the server us missing one or more custom node

The server.log says:

TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
Cannot import /Users/name/Library/Application Support/krita/ai_diffusion/server/ComfyUI/custom_nodes/comfyui-tooling-nodes module for custom nodes: unsupported operand type(s) for |: 'type' and 'NoneType'
(IMPORT FAILED): /Users/name/Library/Application Support/krita/ai_diffusion/server/ComfyUI/custom_nodes/comfyui-tooling-nodes

I deleted everything and let it install again from scratch but ended up with the same error. 
I have "Local Managed Server" in krita and never touched it otherwise.
It is indeed running, I can access it via browser and generate an image. I am not sure, where the error with the custom node comes from (the tooling nodes folder contains files, like the other folders). However, I am not very familiar with comfy, so I don't know what I am looking for.

Thanks for this great tool!

**Comment by Acly** [comment]
It is likely because you are using an older Python version (3.9 maybe).

You might be able to fix it locally by editing `.../Application Support/krita/ai_diffusion/server/ComfyUI/custom_nodes/comfyui-tooling-nodes/translation.py` like here:
https://github.com/Acly/comfyui-tooling-nodes/commit/42965013f9e9d0fe9f4f71efc3654dc8648b34a4#diff-07ea66e72e1d5b170cbe6bdf44844142966b3b4a91e43055ffcb67a6e9277358

**Comment by Doc-Acula** [comment]
Yes, that fixed it. Thank you.
I must have lost track of my python version...it was actually 3.12.3
